2024-07-11 12:54:19.122559: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:54:19.122737: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:54:27,745 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:54:27,833 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:54:27,833 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:54:27,833 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:54:29,242 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:54:29,242 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,258 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.43333333333333335}
INFO flwr 2024-07-11 12:54:29,258 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:54:29,418 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,443 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:54:29,445 | fedavgm.py:198 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,452 | server.py:131 | fit progress: (1, 0.3266284859084374, {'accuracy': 0.8533333333333334}, 0.1932508999999989)
DEBUG flwr 2024-07-11 12:54:29,453 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,473 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:54:29,473 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:54:29,473 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,496 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,506 | server.py:131 | fit progress: (2, 0.32575052033541596, {'accuracy': 0.8566666666666667}, 0.24805839999999968)
DEBUG flwr 2024-07-11 12:54:29,507 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,527 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:54:29,527 | server.py:227 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,548 | server.py:241 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,561 | server.py:131 | fit progress: (3, 0.32574754918328525, {'accuracy': 0.8566666666666667}, 0.3023454999999995)
DEBUG flwr 2024-07-11 12:54:29,561 | server.py:178 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,582 | server.py:192 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:54:29,582 | server.py:227 | fit_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,607 | server.py:241 | fit_round 4 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,618 | server.py:131 | fit progress: (4, 0.3257475747784768, {'accuracy': 0.8566666666666667}, 0.35906729999999953)
DEBUG flwr 2024-07-11 12:54:29,618 | server.py:178 | evaluate_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,639 | server.py:192 | evaluate_round 4 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:54:29,640 | server.py:227 | fit_round 5: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:54:29,674 | server.py:241 | fit_round 5 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,684 | server.py:131 | fit progress: (5, 0.3257475617446477, {'accuracy': 0.8566666666666667}, 0.4247363999999987)
DEBUG flwr 2024-07-11 12:54:29,684 | server.py:178 | evaluate_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:54:29,712 | server.py:192 | evaluate_round 5 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:54:29,713 | server.py:227 | fit_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:54:29,741 | server.py:241 | fit_round 6 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,753 | server.py:131 | fit progress: (6, 0.330777822225389, {'accuracy': 0.8466666666666667}, 0.4944872999999994)
DEBUG flwr 2024-07-11 12:54:29,753 | server.py:178 | evaluate_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:54:29,779 | server.py:192 | evaluate_round 6 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:54:29,779 | server.py:227 | fit_round 7: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:54:29,828 | server.py:241 | fit_round 7 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,840 | server.py:131 | fit progress: (7, 0.3308042442464145, {'accuracy': 0.8466666666666667}, 0.5815895999999992)
DEBUG flwr 2024-07-11 12:54:29,841 | server.py:178 | evaluate_round 7: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:54:29,870 | server.py:192 | evaluate_round 7 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:54:29,870 | server.py:227 | fit_round 8: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:54:29,904 | server.py:241 | fit_round 8 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,916 | server.py:131 | fit progress: (8, 0.33080308704509614, {'accuracy': 0.8466666666666667}, 0.6569894999999999)
DEBUG flwr 2024-07-11 12:54:29,916 | server.py:178 | evaluate_round 8: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:54:29,940 | server.py:192 | evaluate_round 8 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:54:29,940 | server.py:227 | fit_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:54:29,984 | server.py:241 | fit_round 9 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:29,997 | server.py:131 | fit progress: (9, 0.33004826212420935, {'accuracy': 0.8533333333333334}, 0.7387148999999997)
DEBUG flwr 2024-07-11 12:54:29,998 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:54:30,030 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:54:30,030 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:54:30,072 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:54:30,086 | server.py:131 | fit progress: (10, 0.3307873487418797, {'accuracy': 0.8533333333333334}, 0.8268898)
DEBUG flwr 2024-07-11 12:54:30,086 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:54:30,118 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:54:30,118 | server.py:153 | FL finished in 0.8593897999999989
INFO flwr 2024-07-11 12:54:30,119 | app.py:225 | app_fit: losses_distributed [(1, 0.3543968088924885), (2, 0.3543000463396311), (3, 0.35429863817989826), (4, 0.3542986009269953), (5, 0.376852681239446), (6, 0.3748964657386144), (7, 0.3748933672904968), (8, 0.374893123904864), (9, 0.3880760967731476), (10, 0.3879268799509321)]
INFO flwr 2024-07-11 12:54:30,119 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:54:30,119 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:54:30,119 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.3266284859084374), (2, 0.32575052033541596), (3, 0.32574754918328525), (4, 0.3257475747784768), (5, 0.3257475617446477), (6, 0.330777822225389), (7, 0.3308042442464145), (8, 0.33080308704509614), (9, 0.33004826212420935), (10, 0.3307873487418797)]
INFO flwr 2024-07-11 12:54:30,119 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.43333333333333335), (1, 0.8533333333333334), (2, 0.8566666666666667), (3, 0.8566666666666667), (4, 0.8566666666666667), (5, 0.8566666666666667), (6, 0.8466666666666667), (7, 0.8466666666666667), (8, 0.8466666666666667), (9, 0.8533333333333334), (10, 0.8533333333333334)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[-2.27912388,  1.23408997,  0.89899138,  0.4595263 ,  0.24048244,
         0.07484587,  0.03695987, -0.01609832]]), array([0.0076575])]
[array([[-2.26841229,  1.24279797,  0.92187701,  0.46139717,  0.21623002,
         0.06449821,  0.0460821 , -0.02019132]]), array([-0.0160538])]
[array([[-2.26864262,  1.24292437,  0.9218433 ,  0.46163384,  0.21617793,
         0.0645277 ,  0.04617928, -0.02035198]]), array([-0.01605841])]
[array([[-2.2686483 ,  1.24293167,  0.92183879,  0.46163064,  0.21618351,
         0.06452305,  0.04618306, -0.02035145]]), array([-0.01606275])]
[array([[-2.26865099,  1.24293421,  0.92183809,  0.46163049,  0.21618462,
         0.06452192,  0.04618392, -0.02035156]]), array([-0.0160637])]
[array([[-2.19185964,  1.08945256,  0.86581586,  0.42332957,  0.25333092,
         0.08104778, -0.07415603, -0.04714969]]), array([0.19541156])]
[array([[-2.19275665,  1.08925417,  0.86531247,  0.42331663,  0.25343849,
         0.08067798, -0.07471301, -0.04785955]]), array([0.19909705])]
[array([[-2.19280432,  1.08928413,  0.86535108,  0.42337331,  0.25347216,
         0.08067976, -0.07471089, -0.04787001]]), array([0.19900673])]
[array([[-2.13871462,  1.05556948,  0.88325049,  0.43878795,  0.21859681,
         0.07957779, -0.0234866 , -0.0340805 ]]), array([0.04707528])]
[array([[-2.12571152,  1.03856175,  0.85314755,  0.43759248,  0.2337574 ,
         0.06516698,  0.00608724, -0.02961898]]), array([0.05642642])]
>>>>>>>>>>>>>run time:
0.640625s
